### PR TITLE
Chore/refactor tests

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR:
- moves remaining tests into packages
- cleans up fixtures & conftest.py

## Fixes

-

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

-

### CODE REVIEW

File: packages/valory/agents/apy_estimation_chained/README.md [hunk 1]
```diff
@@ -1,4 +0,0 @@
-# ABCI APY Estimation Agent
```
Comment (DavidMinarsch) [inline packages/valory/agents/apy_estimation_chained/README.md:1]: Not used afaik

File: packages/valory/skills/liquidity_provision_abci/tests/test_integration.py [hunk 2]
```diff
@@ -107,7 +112,7 @@ class LiquidityProvisionIntegrationBaseCase(
     exit_nonce: int
     swap_back_nonce: int
     default_synchronized_data_hash: LiquidityRebalancingSynchronizedSata
-    ROOT_DIR = ROOT_DIR
+    ROOT_DIR = Path(".")
```
Comment (DavidMinarsch) [inline packages/valory/skills/liquidity_provision_abci/tests/test_integration.py:115]: tbd

File: packages/valory/skills/oracle_abci/tests/test_integration.py [hunk 3]
```diff
@@ -106,7 +111,7 @@ class TransactionSettlementIntegrationBaseCase(
     """Base case for integration testing TransactionSettlement FSM Behaviour."""
 
     price_estimation_synchronized_data: PriceEstimationSynchronizedSata
-    ROOT_DIR = ROOT_DIR
+    ROOT_DIR = Path(".")
```
Comment (DavidMinarsch) [inline packages/valory/skills/oracle_abci/tests/test_integration.py:114]: tbd

File: packages/valory/skills/oracle_abci/tests/test_integration.py [hunk 4]
```diff
@@ -358,7 +363,7 @@ def _test_same_keeper(self) -> None:
 class TestKeepers(OracleBehaviourBaseCase, IntegrationBaseCase):
     """Test the keepers related functionality for the tx settlement skill."""
 
-    ROOT_DIR = ROOT_DIR
+    ROOT_DIR = Path(".")
```
Comment (DavidMinarsch) [inline packages/valory/skills/oracle_abci/tests/test_integration.py:366]: tbd

File: tests/conftest.py [hunk 5]
```diff
@@ -19,235 +19,28 @@
 
 """Conftest module for Pytest."""
 import inspect
```
Comment (DavidMinarsch) [inline tests/conftest.py:21]: massively simplified

**Review by DavidMinarsch — COMMENTED** [review]
Current mac-os failure:
```
__ ERROR collecting packages/valory/connections/abci/tests/test_abci_fuzz.py ___
tests/test_abci_fuzz.py:30: in <module>
    from packages.valory.connections.abci.tests.helper import (
tests/helper.py:140: in <module>
    AEA_CUSTOM = get_aea_classes(valory_abci_protocol.custom_types)
E   AttributeError: module 'packages.valory.protocols.abci' has no attribute 'custom_types'
__ ERROR collecting packages/valory/connections/abci/tests/test_abci_spec.py ___
tests/test_abci_spec.py:36: in <module>
    from packages.valory.connections.abci.tests.helper import (
tests/helper.py:140: in <module>
    AEA_CUSTOM = get_aea_classes(valory_abci_protocol.custom_types)
E   AttributeError: module 'packages.valory.protocols.abci' has no attribute 'custom_types'
```

**Review by DavidMinarsch — COMMENTED** [review]
`tests/test_deployments/test_app.py::TestTendermintBufferWorking::test_tendermint_buffer` failing on windows

**Review by DavidMinarsch — COMMENTED** [review]
`e2e` tests currently not running - I suspect posargs not forwarded correctly.

File: packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py [hunk 6]
```diff
@@ -37,18 +37,18 @@
 from aea_ledger_ethereum import EthereumCrypto
 from aea_test_autonomy.configurations import ETHEREUM_KEY_DEPLOYER, get_key
 from aea_test_autonomy.docker.base import skip_docker_tests
+from aea_test_autonomy.fixture_helpers import gnosis_safe_hardhat_scope_class
 
 from packages.valory.connections.ledger.connection import LedgerConnection
+from packages.valory.contracts.gnosis_safe.contract import (
+    PUBLIC_ID as GNOSIS_SAFE_PUBLIC_ID,
+)
```
Comment (angrybayblade) [inline packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py:45]: Moved this to `safe_deployment_abci`

File: packages/valory/connections/ledger/tests/test_contract_api.py [hunk 7]
```diff
@@ -35,16 +39,15 @@
...
-    PUBLIC_ID as GNOSIS_SAFE_PUBLIC_ID,
-)
 from packages.valory.protocols.contract_api.dialogues import ContractApiDialogue
 from packages.valory.protocols.contract_api.dialogues import (
     ContractApiDialogues as BaseContractApiDialogues,
 )
 from packages.valory.protocols.contract_api.message import ContractApiMessage
 
 
+GNOSIS_SAFE_PUBLIC_ID = PublicId.from_str("valory/gnosis_safe:0.1.0")
```
[the remark this answers is not in the document]
Comment (angrybayblade) [reply packages/valory/connections/ledger/tests/test_contract_api.py]: https://github.com/valory-xyz/open-autonomy/pull/1325/commits/e5ea726647b8ad385c726547c3b3dca8341b6601

File: packages/valory/skills/oracle_abci/tests/test_integration.py (hunk 4 above)
Comment (angrybayblade) [reply packages/valory/skills/oracle_abci/tests/test_integration.py]: https://github.com/valory-xyz/open-autonomy/pull/1325/commits/c0afd4a6f36efa67a1c2c7f0ed43e2821272f7df

File: packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py (hunk 6 above)
Comment (DavidMinarsch) [reply packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py]: Why?

Comment (DavidMinarsch) [reply packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py]: I moved it here on purpose

File: packages/valory/skills/liquidity_provision_abci/tests/test_integration.py [hunk 8]
```diff
@@ -115,7 +115,6 @@ class LiquidityProvisionIntegrationBaseCase(
     exit_nonce: int
     swap_back_nonce: int
     default_synchronized_data_hash: LiquidityRebalancingSynchronizedSata
-    ROOT_DIR = Path(".")
```
[the remark this answers is not in the document]
Comment (DavidMinarsch) [reply packages/valory/skills/liquidity_provision_abci/tests/test_integration.py:118]: This was a placeholder

File: packages/valory/skills/oracle_abci/tests/test_integration.py [hunk 9]
```diff
@@ -115,7 +115,6 @@ class TransactionSettlementIntegrationBaseCase(
     """Base case for integration testing TransactionSettlement FSM Behaviour."""
 
     price_estimation_synchronized_data: PriceEstimationSynchronizedSata
-    ROOT_DIR = Path(".")
```
Comment (DavidMinarsch) [inline packages/valory/skills/oracle_abci/tests/test_integration.py:118]: see above

File: packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py [hunk 10]
```diff
@@ -39,16 +38,15 @@
...
 from packages.valory.protocols.contract_api.dialogues import (
     ContractApiDialogues as BaseContractApiDialogues,
 )
 from packages.valory.protocols.contract_api.message import ContractApiMessage
-
-
-GNOSIS_SAFE_PUBLIC_ID = PublicId.from_str("valory/gnosis_safe:0.1.0")
-
-SOME_SKILL_ID = "some/skill:0.1.0"
+from packages.valory.skills.safe_deployment_abci import PUBLIC_ID as SKILL_ID
```
Comment (DavidMinarsch) [inline packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py:49]: This is not needed for the test to work...

File: packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py [hunk 11]
```diff
@@ -39,16 +38,15 @@
...
+    PUBLIC_ID as GNOSIS_SAFE_PUBLIC_ID,
+)
 from packages.valory.protocols.contract_api.dialogues import ContractApiDialogue
 from packages.valory.protocols.contract_api.dialogues import (
     ContractApiDialogues as BaseContractApiDialogues,
 )
 from packages.valory.protocols.contract_api.message import ContractApiMessage
-
-
-GNOSIS_SAFE_PUBLIC_ID = PublicId.from_str("valory/gnosis_safe:0.1.0")
```
Comment (DavidMinarsch) [inline packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py:49]: The test works just fine like this. We can then fix it properly another time. I prefer it to stay in `ledger` connection as this actually tests that.

File: packages/valory/skills/liquidity_provision_abci/tests/test_integration.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (angrybayblade) [reply packages/valory/skills/liquidity_provision_abci/tests/test_integration.py:118]: Yes, I made sure it was safe to remove

File: packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py (hunk 11 above)
Comment (angrybayblade) [reply packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py:49]: So my problem with this being in the ledger connection package is, it still depends on the contract which not one of the dependency for the ledger connection. Another thing is we need to import and patch the `GnosisSafeContract` class for so many of the tests to work, which is not possible unless we define the contract as a dependency for the connection.

Take a look here you'll understand  https://github.com/valory-xyz/open-autonomy/pull/1329/files#diff-693e5a8d749782dd6ee62d30aad262db4c0a79c255704e32a24c588f675980fcL310-R332

Comment (DavidMinarsch) [reply packages/valory/skills/safe_deployment_abci/tests/test_contract_api.py:49]: Ok

File: packages/valory/connections/ledger/tests/conftest.py [hunk 12]
```diff
@@ -58,17 +52,11 @@
 
 PACKAGE_DIR = Path(__file__).parent.parent
 
-ETHEREUM_DEFAULT_LEDGER_CONFIG = {
-    "address": f"{DEFAULT_GANACHE_ADDR}:{DEFAULT_GANACHE_PORT}",
-    "chain_id": DEFAULT_GANACHE_CHAIN_ID,
-    # "denom": ETHEREUM_DEFAULT_CURRENCY_DENOM, # noqa: E800
-    # "gas_price_api_key": GAS_PRICE_API_KEY, # noqa: E800
-    "default_gas_price_strategy": "eip1559",
```
[the remark this answers is not in the document]
Comment (DavidMinarsch) [reply packages/valory/connections/ledger/tests/conftest.py:66]: I suspect all test failures in that issue have the same reason

**Pull request merged**